frontend/class-public.php

```diff
@@ -177,41 +177,56 @@ public function volunteer_opps_register_block(){
 			'all'
 		);
 
-		register_block_type( 'wired-impact-volunteer-management/volunteer-opps', array(
-				'editor_script' 	=> $this->plugin_name . '-block',
-				'editor_style'		=> $this->plugin_name,
+		register_block_type(
+			'wired-impact-volunteer-management/volunteer-opps',
+			array(
+				'editor_script'   => $this->plugin_name . '-block',
+				'editor_style'    => $this->plugin_name,
 				'render_callback' => array( $this, 'display_volunteer_opps_block' ),
 				'attributes'      => array(
-					'showOneTime'    => array(
-							'type'      => 'boolean',
-							'default'   => true,
+					'showOneTime' => array(
+						'type'    => 'boolean',
+						'default' => true,
 					),
-			),
-		) );
+					'className'   => array(
+						'type'    => 'string',
+						'default' => '',
+					),
+					'anchor'      => array(
+						'type'      => 'string',
+						'default'   => '',
+						'attribute' => 'id',
+						'selector'  => '.volunteer-opps',
+					),
+				),
+			)
+		);
 	}
 
 	/**
-	 * Display our volunteer opportunities block in the admin or 
+	 * Display our volunteer opportunities block in the admin or
 	 * on the frontend of the website. AJAX is used to load the
 	 * volunteer opportunities in the admin.
-	 * 
+	 *
 	 * @see    https://wordpress.org/gutenberg/handbook/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks/
-	 * @param  array 	$attributes	Attributes saved from the block editor to display in the admin or frontend.
-	 * @return string        			The final HTML for our volunteer opportunities.
+	 * @param  array $attributes        Attributes saved from the block editor to display in the admin or frontend.
+	 * @return string                   The final HTML for our volunteer opportunities.
 	 */
-	public function display_volunteer_opps_block( $attributes ){
-	
-		if( $attributes['showOneTime'] === true ){
-			return $this->display_one_time_volunteer_opps();
-		} 
-		
-		return $this->display_flexible_volunteer_opps();
+	public function display_volunteer_opps_block( $attributes ) {
+
+		if ( $attributes['showOneTime'] === true ) {
+			return $this->display_one_time_volunteer_opps( $attributes );
+		}
+
+		return $this->display_flexible_volunteer_opps( $attributes );
 	}
 
 	/**
 	 * Shortcode for viewing all one-time volunteer opportunities.
+	 *
+	 * @param array $block_attributes  Attributes saved from the block editor to display in the admin or frontend.
 	 */
-	public function display_one_time_volunteer_opps(){
+	public function display_one_time_volunteer_opps( $block_attributes ) {
```

Comment:
Should we update the `$block_attributes` variable name to be `$attributes` since this isn't just block attributes. If a shortcode is used instead they are shortcode attributes. If we do make this change I imagine we'll need to make it in a few places and also update comments throughout the file.

Reply:
Yeah, definitely, great call. I'll get this updated throughout the file.

Reply:
Looks like sanitize_text_field() works well here!

Reply:
These have been updated.